crates/host/src/wasmbus/handler.rs

```diff
@@ -456,6 +460,642 @@ impl Messaging for Handler {
+                        match *message {
+                            Message::Nats(async_nats::Message {
+                                payload,
+                                headers: Some(headers),
+                                ..
+                            }) => nats.publish_with_headers(topic, headers, payload).await,
+                            Message::Nats(async_nats::Message { payload, .. }) => {
+                                nats.publish(topic, payload).await
+                            }
+                        }
+                    }
+                    messaging0_3_0::types::Message::Wrpc(messaging::types::BrokerMessage {
+                        body,
+                        ..
+                    }) => nats.publish(topic, body).await,
+                    messaging0_3_0::types::Message::Guest(MessagingGuestMessage0_3 {
+                        content_type,
+                        data,
+                        metadata,
+                    }) => {
+                        if content_type.is_some() {
+                            return Ok(Err(messaging0_3_0::types::Error::Other(
+                                "`content-type` not supported by NATS.io".into(),
+                            )));
+                        }
+                        if let Some(metadata) = metadata {
+                            nats.publish_with_headers(
+                                topic,
+                                metadata
+                                    .into_iter()
+                                    .map(|(k, v)| (k.into_header_name(), v.into_header_value()))
+                                    .collect(),
+                                data.into(),
+                            )
+                            .await
+                        } else {
+                            nats.publish(topic, data.into()).await
+                        }
+                    }
+                } {
+                    Ok(()) => return Ok(Ok(())),
+                    Err(err) => {
+                        // TODO: Correctly handle error kind
+                        return Ok(Err(messaging0_3_0::types::Error::Other(err.to_string())));
+                    }
+                }
+            }
+            let body = match message {
+                messaging0_3_0::types::Message::Host(message) => {
+                    let message = message
+                        .into_any()
+                        .downcast::<Message>()
+                        .map_err(|_| anyhow!("unknown message type"))?;
+                    match *message {
+                        Message::Nats(async_nats::Message {
+                            headers: Some(..), ..
+                        }) => {
+                            return Ok(Err(messaging0_3_0::types::Error::Other(
+                                "headers not currently supported by wRPC targets".into(),
+                            )));
```

Comment:
This seems like a pretty common case to use headers on the message for 🤔 What's the reasoning for this? Is it the same as why we need to specify headers in the client in that it adjusts the payload size for wRPC?

Reply:
`wasmcloud:messaging@0.2.0`, used on wRPC level, does not support headers: https://github.com/wasmCloud/messaging/blob/ff300523cfdaf5eab19a9385637ae4665cc47d86/wit/messaging.wit#L1-L27

If we want to use wRPC for binary providers, then we'll need a `wrpc:messaging` defined first, which would include support for headers